src/add_prefix.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright 2017 Kisio Digital and/or its affiliates.
+//
+// This program is free software: you can redistribute it and/or
+// modify it under the terms of the GNU General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see
+// <http://www.gnu.org/licenses/>.
+
+//! A trait for every structure that needs to be updated with a prefix
+
+use crate::{
+    collection::{Collection, CollectionWithId, Id},
+    model::Collections,
+};
+
+pub trait AddPrefix {
+    fn add_prefix(&mut self, prefix: &str);
+    fn add_prefix_with_sep(&mut self, prefix: &str, sep: &str) {
+        let prefix = format!("{}{}", prefix, sep);
+        self.add_prefix(&prefix);
+    }
+}
+
+impl<T> AddPrefix for Collection<T>
+where
+    T: AddPrefix,
+{
+    fn add_prefix(&mut self, prefix: &str) {
+        for obj in &mut self.values_mut() {
+            obj.add_prefix(prefix);
+        }
+    }
+}
+
+impl<T> AddPrefix for CollectionWithId<T>
+where
+    T: Id<T> + AddPrefix,
+{
+    fn add_prefix(&mut self, prefix: &str) {
+        let indexes: Vec<_> = self.get_id_to_idx().values().cloned().collect();
```

Comment:
I wonder if it's better like that

```suggestion
        let indexes: Vec<_> = self.iter().map(|(idx, _)| idx).collect();
```

Reply:
Yes, I think it is for 2 reasons:
- No explicit clone (the copy of `Idx` happened in the `iter()` though)
- Not using this internal map `id_to_idx` which I'm not even sure why we expose it through the function `get_id_to_idx()`

Thank you :smiley: